framework/src/base/FEProblem.C

```diff
@@ -1575,6 +1577,34 @@ FEProblem::projectSolution()
 }
 
 
+MooseSharedPointer<DiscreteMaterial>
+FEProblem::getDiscreteMaterial(std::string name, Moose::MaterialDataType type, THREAD_ID tid)
+{
+  MooseSharedPointer<Material> output;
+  switch (type)
+  {
+  case Moose::BLOCK_MATERIAL_DATA:
+    output = _materials.getObject(name, tid);
+    break;
+  case Moose::NEIGHBOR_MATERIAL_DATA:
+    name += "_neighbor";
+    output = _neighbor_materials.getObject(name, tid);
+    break;
+  case Moose::BOUNDARY_MATERIAL_DATA:
+  case Moose::FACE_MATERIAL_DATA:
+    name += "_face";
+    output = _face_materials.getObject(name, tid);
+    break;
+  }
+
+  MooseSharedPointer<DiscreteMaterial> discrete = MooseSharedNamespace::dynamic_pointer_cast<DiscreteMaterial>(output);
+  if (!discrete)
+    mooseError("Unable to locate the a DiscreteMaterial with the name " << name);
```

Comment:
Ah - so here's where we'll hit the mooseError(). You might want to document that it's possible to get an empty pointer though in your method.